src/Chainweb/Pact/PactService/Checkpointer.hs

```diff
@@ -349,7 +349,12 @@ rewindTo rewindLimit (Just (ParentHeader parent)) = do
         -- This does a restore, i.e. it rewinds the checkpointer back in
         -- history, if needed.
         withCheckpointerWithoutRewind (Just target) "fastForward" $ \pdbenv -> do
-            payload <- liftIO (payloadWithOutputsToPayloadData <$> casLookupM payloadDb bpHash)
+            payload <- liftIO $ casLookup payloadDb bpHash >>= \case
+                Nothing -> throwM $ PactInternalError
```

Comment:
This change is for a more localized error yes?

Reply:
Yes, it was helpful during debugging. Before it was throwing just an generic message coming from the Chainweb-storage package.